src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -1670,6 +1867,40 @@ private void ResponseCallback(IAsyncResult asynchronousResult)
             }
         }
 
+        private static async Task VerifyNewConnection(string responseContent, Task<Socket> secondAccept, Task<WebResponse> currentResponseTask)
+        {
+            var secondServer = await secondAccept;
+            Assert.True(secondAccept.IsCompleted);
+            using (var secondStream = new NetworkStream(secondServer, ownsSocket: false))
+            using (var secondReader = new StreamReader(secondStream))
+            {
+                await ReplyToClient(responseContent, secondServer, secondReader);
+                await VerifyResponse(responseContent, currentResponseTask);
+            }
+        }
+
+        private static async Task ReplyToClient(string responseContent, Socket server, StreamReader serverReader)
+        {
+            string responseBody =
+                    "HTTP/1.1 200 OK\r\n" +
+                    $"Date: {DateTimeOffset.UtcNow:R}\r\n" +
+                    $"Content-Length: {responseContent.Length}\r\n" +
+                    "\r\n" + responseContent;
+            while (!string.IsNullOrWhiteSpace(await serverReader.ReadLineAsync())) ;
+            await server.SendAsync(new ArraySegment<byte>(Encoding.ASCII.GetBytes(responseBody)), SocketFlags.None);
+        }
+
+        private static async Task VerifyResponse(string expectedResponse, Task<WebResponse> responseTask)
+        {
+            var firstRequest = await responseTask;
+            using (var firstResponseStream = firstRequest.GetResponseStream())
+            using (var reader = new StreamReader(firstResponseStream))
+            {
+                var response = reader.ReadToEnd();
```

Comment:
Nit: please only use var when the type is evident on the right-hand side (ctors, explicit casts)